lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -844,15 +815,112 @@ private void indexObjectWrapperFn(final List<Object> input)
         
         private List<ObjectData> getObjectsByIds(final Set<GUID> guids)
                 throws RetriableIndexingException {
-            kbasesearchengine.search.PostProcessing pp = 
-                    new kbasesearchengine.search.PostProcessing();
-            pp.objectData = false;
-            pp.objectKeys = true;
-            try {
-                return indexingStorage.getObjectsByIds(guids, pp);
-            } catch (IOException e) {
-                throw new RetriableIndexingException(ErrorType.OTHER, e.getMessage(), e);
+
+            final List<ObjectData> data = new ArrayList<>();
+
+            // get object from data source
+            for (GUID guid: guids) {
+                final File tempFile;
+                try {
+                    FileUtil.getOrCreateSubDir(rootTempDir, guid.getStorageCode());
+                    tempFile = File.createTempFile("ws_srv_response_", ".json");
+                } catch (IOException ex) {
+                    throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage(), ex);
+                }
+
+                try {
+                    final EventHandler handler = getEventHandler(guid);
+                    final Iterator<ResolvedReference> refs = handler.resolveReferences(null, guids).iterator();
+                    while (refs.hasNext()) {
+                        final ResolvedReference ref = refs.next();
+
+                        // make a copy to avoid mutating the caller's path
```

Comment:
But you're not using a path here? The equivalent to the next 3 lines is:
`final SourceData obj = handler.load(Arrays.asList(guid), tempFile.toPath());`

Reply:
Added refpath but I don't know if this is right or wrong as I don't understand the resolved path concept fully. Please check.